org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/InferenceContext18.java

```diff
@@ -1130,151 +1147,8 @@ public boolean reduceAndIncorporate(ConstraintFormula constraint) throws Inferen
-							for (int j = 0; j < numVars; j++)
-								if (TypeBinding.equalsEquals(variables[j], typeVariable))
-									return zs[j];
-							/* If we have an instantiation, lower it to the instantiation. We don't want downstream abstractions to be confused about multiple versions of bounds without
-							   and with instantiations propagated by incorporation. See https://bugs.eclipse.org/bugs/show_bug.cgi?id=430686. There is no value whatsoever in continuing
-							   to speak in two tongues. Also fixes https://bugs.eclipse.org/bugs/show_bug.cgi?id=425031.
-							*/
-							if (typeVariable instanceof InferenceVariable) {
-								InferenceVariable inferenceVariable = (InferenceVariable) typeVariable;
-								TypeBinding instantiation = kurrentBoundSet.getInstantiation(inferenceVariable, null);
-								if (instantiation != null)
-									return instantiation;
-							}
-							return typeVariable;
-						}
-					};
-					for (int j = 0; j < numVars; j++) {
-						InferenceVariable variable = variables[j];
-						CaptureBinding18 zsj = zs[j];
-						// add lower bounds:
-						TypeBinding[] lowerBounds = tmpBoundSet.lowerBounds(variable, true/*onlyProper*/);
-						if (lowerBounds != Binding.NO_TYPES) {
-							TypeBinding lub = this.scope.lowerUpperBound(lowerBounds);
-							if (lub != TypeBinding.VOID && lub != null)
-								zsj.lowerBound = lub;
-						}
-						// add upper bounds:
-						TypeBinding[] upperBounds = tmpBoundSet.upperBounds(variable, false/*onlyProper*/);
-						if (upperBounds != Binding.NO_TYPES) {
-							for (int k = 0; k < upperBounds.length; k++)
-								upperBounds[k] = Scope.substitute(theta, upperBounds[k]);
-							if (!setUpperBounds(zsj, upperBounds))
-								continue; // at violation of well-formedness skip this candidate and proceed
-						}
-						if (tmpBoundSet == this.currentBounds)
-							tmpBoundSet = tmpBoundSet.copy();
-						Iterator<ParameterizedTypeBinding> captureKeys = tmpBoundSet.captures.keySet().iterator();
-						Set<ParameterizedTypeBinding> toRemove = new LinkedHashSet<ParameterizedTypeBinding>();
-						while (captureKeys.hasNext()) {
-							ParameterizedTypeBinding key = captureKeys.next();
-							int len = key.arguments.length;
-							for (int i = 0; i < len; i++) {
-								if (TypeBinding.equalsEquals(key.arguments[i], variable)) {
-									toRemove.add(key);
-									break;
-								}
-							}
-						}
-						captureKeys = toRemove.iterator();
-						while (captureKeys.hasNext())
-							tmpBoundSet.captures.remove(captureKeys.next());
-						tmpBoundSet.addBound(new TypeBound(variable, zsj, ReductionResult.SAME), this.environment);
-					}
-					if (tmpBoundSet.incorporate(this)) {
-						if (tmpBoundSet.numUninstantiatedVariables(this.inferenceVariables) == oldNumUninstantiated)
-							return null; // abort because we made no progress
-						continue;
-					}
-					return null;
-				}
-			}
-		}
-		return tmpBoundSet;
+		return resolve(toResolve, false);
```

Comment:
please avoid moving code around like this, it would kill this method's git history which is vital for future maintenance!